src/Swarm/Game/State.hs

```diff
@@ -440,7 +445,7 @@ viewCenterRule = lens getter setter
     case rule of
       VCLocation v2 -> g {_viewCenterRule = rule, _viewCenter = v2}
       VCRobot rid ->
-        let robotcenter = g ^? robotMap . ix rid <&> view robotLocation
+        let robotcenter = g ^? robotMap . ix rid . robotLocation
          in -- retrieve the loc of the robot if it exists, Nothing otherwise.
             -- sometimes, lenses are amazing...
             case robotcenter of
```

Comment:
_Sometimes, lenses are amazing..._ and sometimes they give you a headache, trying to find out why is the special robot that you added (later) to the game state not focused. :smile:

Reply:
The lenses I used were ordered with `.` not `&`, so when I added a robot on line X and called this lens on line X+1 it did not do anything.

It did not really matter, because there were other issues, so it was more of a funny discovery 😄